lib/webauthn/public_key_credential/creation_options.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require "cose/algorithm"
+require "webauthn/public_key_credential/options"
+require "webauthn/public_key_credential/rp_entity"
+require "webauthn/public_key_credential/user_entity"
+
+module WebAuthn
+  # TODO: make keyword arguments mandatory in next major version
+  def self.credential_creation_options(rp_name: nil, user_name: "web-user", display_name: "web-user", user_id: "1")
+    warn(
+      "DEPRECATION WARNING: `WebAuthn.credential_creation_options` is deprecated."\
+      " Please use `WebAuthn::PublicKeyCredential.create_options` instead."
+    )
+
+    PublicKeyCredential::CreationOptions.new(
+      rp: { name: rp_name },
+      user: { id: user_id, name: user_name, display_name: display_name }
+    ).as_json
+  end
+
+  class PublicKeyCredential
+    def self.create_options(options, encoding: :base64url)
+      WebAuthn::PublicKeyCredential::CreationOptions.new(options.merge(encoding: encoding))
+    end
+
+    class CreationOptions < Options
+      attr_accessor(
+        :attestation,
+        :authenticator_selection,
+        :exclude,
+        :algs,
+        :rp,
+        :user
+      )
+
+      def initialize(
+        attestation: nil,
+        authenticator_selection: nil,
+        exclude_credentials: nil,
+        exclude: nil,
+        pub_key_cred_params: nil,
+        algs: nil,
+        rp: {}, # rubocop:disable Naming/UncommunicativeMethodParamName
```

Comment:
Given our problem domain let's add `rp` to `AllowedNames` so Rubocop will stop complaining about this 😄 https://rubocop.readthedocs.io/en/latest/cops_naming/#naminguncommunicativemethodparamname

Reply:
Good call.

Wasn't aware of that configurable attributes.
THanks.